src/dbinterface.jl

```diff
@@ -384,9 +384,17 @@ Base.IteratorEltype(::Type{Cursor{false, knownlength}}) where {knownlength} = Ba
 Base.eltype(x::Cursor{false, knownlength}) where {knownlength} = Row
 
 function Base.iterate(x::Cursor{false}, st=1)
-    status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
-    while status == API.SQL_STILL_EXECUTING
+    while true
         status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
+        if status == API.SQL_SUCCESS_WITH_INFO
+            @warn(API.diagnostics(x.stmt))
+            break
+        end
+        if status == API.SQL_ERROR || status == API.SQL_INVALID_HANDLE
+            error(API.diagnostics(x.stmt))
+        end
+        status == API.SQL_STILL_EXECUTING && continue
+        break
```

Comment:
```suggestion
        status == API.SQL_STILL_EXECUTING || break
```
feels a little cleaner/simpler

Reply:
Indeed!